servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultServiceDiscoveryRetryStrategy.java

```diff
@@ -231,33 +232,41 @@ public boolean isAvailable() {
             return retainedAddresses.isEmpty() ? null : retainedAddresses.values();
         }
 
-        @Nullable
-        Collection<E> consume(final Collection<E> events) {
+        Collection<E> consumeAndFilter(final Collection<E> events) {
             if (retainedAddresses == NONE_RETAINED) {
+                List<E> toReturn = new ArrayList<>(events.size());
                 for (E e : events) {
                     if (e.isAvailable()) {
-                        activeAddresses.put(e.address(), e);
+                        if (activeAddresses.putIfAbsent(e.address(), e) == null) {
+                            // FIXME: what if it's a custom E that has more fields?
+                            toReturn.add(e);    // filter out duplicated addresses
+                        }
                     } else {
-                        activeAddresses.remove(e.address());
+                        if (activeAddresses.remove(e.address()) != null) {
+                            toReturn.add(e);   // filter out addresses that were already unavailable
+                        }
                     }
+                    // FIXME: remove addresses that are available and unavailable in the same events collection
                 }
-                return events;
+                return toReturn;
             }
 
             // we have seen an error, replace cache with new addresses and deactivate the ones which are not present
             // in the new list.
             for (E event : events) {
-                final R address = event.address();
```

Comment:
What is this change for?

Reply:
just reformatting as this variable is not necessary, I can revert

Reply:
Not required, either way is fine. Thought I must be missing some significance.